addons/api/addon/abilities/target.js

```diff
@@ -34,10 +34,25 @@ export default class TargetAbility extends ModelAbility {
     return this.hasAuthorizedAction('add-credential-sources');
   }
 
+  /**
+   * @type {boolean}
+   */
+  get canAddInjectedApplicationCredentialSources() {
+    return (
+      !this.model.isTCP && this.hasAuthorizedAction('add-credential-sources')
+    );
+  }
   /**
    * @type {boolean}
    */
   get canRemoveBrokeredCredentialSources() {
     return this.hasAuthorizedAction('remove-credential-sources');
   }
+
+  /**
+   * @type {boolean}
+   */
+  get canRemoveInjectedApplicationCredentialSources() {
```

Comment:
Do we need different abilities that check the same permissions? Can we call them something more generic?

Reply:
Good point. I'll try making it generic

Reply:
We are adding the ability to check the target type as we currently have no way of differentiating between brokered vs. application credentials. And making it generic will prevent me from checking the type such as `@model.isSSH` and I believe it is preferred to keep this logic within the ability instead of bringing it to the template/components.

Reply:
I might just not know the context but if they both check for the same permission why do you need to differentiate between the types? 

And if we do need the type, is there a reason why `canRemoveInjectedApplicationCredentialSources` isn't also only for ssh targets as well? Shouldn't the add and remove be symmetrical?

Reply:
Add and remove are asymmetrical.  The reason is that only SSH targets support _adding_ injected credentials in the first place.  So you can't have injected credentials on a target you're not supposed to.  Or stated in the positive:  if you have the remove action AND a credential source is assigned to a target, you can definitely remove it regardless of type.  Thus, you can express remove generically, but not add.  The question then becomes should we _treat as symmetrical_ add and remove here.  It's more of a style/philosophical question.

Reply:
The backend has the same permission for both brokered and injected credentials. But currently, only SSH targets can have both brokered and injected.  Adding a type check to `remove` seems unnecessary as the user can't delete a resource they never added(in the case of injected credentials for TCP targets). 

Had an offline discussion with Randy; I can make `remove` generic as we don't care about types for that action.

Reply:
Makes sense, thanks for clarifying!